pkg/mesh/external/tcmesh/types/config/mesh_config.go

```diff
@@ -0,0 +1,256 @@
+	UseNormalJwt bool `json:"useNormalJwt,omitempty"`
+}
+
+// PilotConfiguration defines config options for Pilot
+type PilotConfiguration struct {
+	TraceSampling float32 `json:"traceSampling"`
+
+	CommonConfiguration
+}
+
+// CitadelConfiguration defines config options for Citadel
+type CitadelConfiguration struct {
+	CommonConfiguration
+	SelfSigned bool `json:"selfSigned,omitempty"`
+}
+
+// GalleyConfiguration defines config options for Galley
+type GalleyConfiguration struct {
+	CommonConfiguration
+}
+
+// GatewaysConfiguration defines config options for Gateways
+type GatewaysConfiguration struct {
+	IngressConfigs []IngressGatewayConfiguration `json:"ingressGateways,omitempty"`
+	EgressConfigs  []GatewayConfiguration        `json:"egressGateways,omitempty"`
+}
+
+type GatewaySDSConfiguration struct {
+	Enabled bool   `json:"enabled,omitempty"`
+	Image   string `json:"image,omitempty"`
+}
+
+type GatewayConfiguration struct {
+	Name               string                  `json:"name"`
+	Namespace          string                  `json:"namespace"`
+	ServiceType        v1.ServiceType          `json:"serviceType,omitempty"`
+	ServiceAnnotations map[string]string       `json:"serviceAnnotations,omitempty"`
+	ServiceLabels      map[string]string       `json:"serviceLabels,omitempty"`
+	SDS                GatewaySDSConfiguration `json:"sds,omitempty"`
+	Inner              bool                    `json:"inner,omitempty"` // add label `managed-by: tke-mesh`, not editable
+
+	CommonConfiguration
+}
+
+type IngressGatewayConfiguration struct {
+	GatewayConfiguration `protobuf:"bytes,21,opt,name=gatewayConfiguration"`
+	ExistLBID            string     `json:"existLBID,omitempty" protobuf:"bytes,20,opt,name=resources"` // TCNP-TODO
+	LBName               string     `json:"lbName,omitempty"`                                           // TCNP-TODO
+	AccessType           AccessType `json:"accessType"`                                                 // TCNP-TODO
+	Subnet               string     `json:"subnet,omitempty"`                                           // TCNP-TODO
+}
+
+type K8SIngressConfiguration struct {
+	Disabled bool `json:"disabled,omitempty" protobuf:"varint,1,opt,name=disabled"`
+}
+
+// SidecarInjectorConfiguration defines config options for SidecarInjector
+type SidecarInjectorConfiguration struct {
+	CommonConfiguration
+	// If true, sidecar injector will rewrite PodSpec for liveness
+	// health check to redirect request to sidecar. This makes liveness check work
+	// even when mTLS is enabled.
+	RewriteAppHTTPProbe bool `json:"rewriteAppHTTPProbe,omitempty" protobuf:"varint,3,opt,name=rewriteAppHTTPProbe"`
+}
+
+// MixerConfiguration defines config options for Mixer
+type MixerConfiguration struct {
+	CommonConfiguration
+}
+
+// ProxyConfiguration defines config options for Proxy
+type ProxyConfiguration struct {
+	Image string `json:"image,omitempty" protobuf:"bytes,1,opt,name=image"`
```

Comment:
Some fileds have protobuf tag, but some fileds don't. Why?

Reply:
Legacy code, I will delete the protobuf tags.